### ISSUE TITLE
Signup requires full read & write access for public and private repos

### ISSUE THREAD

**Issue by hadley** [wercker/support#20]
The current requested permissions are excessive and prevent me from trying out werker.

**Comment by craigcitro** [comment]
:+1:

**Comment by hatchan** [comment]
Unfortunately GitHub doesn't allow read only access to private repositories https://help.github.com/articles/connecting-with-third-party-applications/#types-of-application-access-and-data

**Comment by hadley** [comment]
This is a total deal breaker - you would be better off not supporting private repos than asking for complete read and write access to EVERYTHING. If it is absolutely necessary for werker to have this access, you need to precisely why it is needed, and why we should trust you to not abuse the privilege, and how you are keeping the OAuth credentials secure. Additionally, the page that you linked to does not saying anything about limiting read-only access to private repos.

**Comment by hadley** [comment]
Why do you need more access than `user` and `repo_deployment`? (https://developer.github.com/v3/oauth/)

**Comment by jimhester** [comment]
I understand that if you want to use wercker with _private_ repositories you need full read and write access, however if you restrict to only working with _public_ repositories it is possible to do with fewer permissions.  Couldn't wercker default to only working with public repositories with an optional opt-in if the user also wants to use private repositories?

**Comment by hatchan** [comment]
@jimhester this is something which we're going to explore. I cannot give a time indication if or when this is going to change

**Comment by tdhock** [comment]
+1

**Comment by HenrikBengtsson** [comment]
Joining in: Please consider non-write req's for public repositories, which really is a show stopper.

**Comment by cpsievert** [comment]
:+1:

**Comment by krlmlr** [comment]
Any updates on this issue?

**Comment by yoshuawuyts** [comment]
You're right, this is something we could be doing better so I'm reopening this issue. We're currently hard at work on other aspects of `wercker`, so I'm not sure when this will be done. Feel free to ping us on the status though, we're happy to answer any questions you have :tada:

**Comment by gaborcsardi** [comment]
Well, I am not sure who accepts these permissions, but this is clearly not right. Even GitHub warns against it: https://help.github.com/articles/connecting-with-third-party-applications/#repositories

> This is one of the most expansive configurations you can permit, so accept it with caution!

**Comment by krlmlr** [comment]
Thanks for keeping up the good work. Did you have a chance to look into this permission issue? I imagine many potential users are scared away by the permissions that wercker requires by default.
